File: iOS/DoraemonKitDemo/DoraemonKitDemo/DemoVC/Home/DoraemonDemoHomeViewController.m
//
//  DoraemonDemoHomeViewController.m
//  DoraemonKitDemo
//
//  Created by yixiang on 2018/5/15.
//  Copyright © 2018年 yixiang. All rights reserved.
//

#import "DoraemonDemoHomeViewController.h"
#import "DoraemonDemoLoggerViewController.h"
#import "DoraemonDemoPerformanceViewController.h"
#import "DoraemonDemoSanboxViewController.h"
#import "DoraemonDemoUIViewController.h"
#import "DoraemonDemoNetViewController.h"
#import "DoraemonDemoMockGPSViewController.h"
#import "DoraemonDemoCrashViewController.h"
#import "DoraemonDemoCommonViewController.h"
#import <objc/runtime.h>
#import "UIView+Doraemon.h"
#import "UIViewController+Doraemon.h"
#import "DoraemonDemoMemoryLeakViewController.h"
#import "DoraemonDemoMultiControlViewController.h"

@interface DoraemonDemoHomeViewController ()<UITableViewDelegate,UITableViewDataSource>

@property (strong, nonatomic) UITableView *tableView;

@end

@implementation DoraemonDemoHomeViewController


- (UITableView *)tableView {
    if (!_tableView) {
        _tableView = [[UITableView alloc] initWithFrame:self.view.bounds];
        _tableView.delegate = self;
        _tableView.dataSource = self;
    }
    
    return _tableView;
}


- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = DoraemonDemoLocalizedString(@"DoraemonKit");
    self.navigationItem.leftBarButtonItems = nil;
    [self.view addSubview:self.tableView];
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return 11;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    static NSString *cellId = @"cellId";
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:cellId];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellId];
    }
    NSString *txt = nil;
    NSInteger row = indexPath.row;
    if (row==0) {
        txt = DoraemonDemoLocalizedString(@"沙盒测试Demo");
    }else if(row==1){
        txt = DoraemonDemoLocalizedString(@"日志测试Demo");
    }else if(row==2){
        txt = DoraemonDemoLocalizedString(@"性能测试Demo");
    }else if(row==3){
        txt = DoraemonDemoLocalizedString(@"视觉测试Demo");
    }else if(row==4){
        txt = DoraemonDemoLocalizedString(@"网络测试Demo");
    }else if(row==5){
        txt = DoraemonDemoLocalizedString(@"模拟位置Demo");
    }else if(row==6){
        txt = DoraemonDemoLocalizedString(@"crash触发Demo");
    }else if(row==7){
        txt = DoraemonDemoLocalizedString(@"通用测试Demo");
    }else if(row==8){
        txt = DoraemonDemoLocalizedString(@"内存泄漏测试");
    }else if(row == 9){
        txt = DoraemonDemoLocalizedString(@"一机多控测试Demo");
    }
    cell.textLabel.text = txt;
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    NSInteger row = indexPath.row;
    UIViewController *vc = nil;
    if (row == 0) {
        vc = [[DoraemonDemoSanboxViewController alloc] init];
    }else if(row == 1){
        vc = [[DoraemonDemoLoggerViewController alloc] init];
    }else if(row == 2){
        vc = [[DoraemonDemoPerformanceViewController alloc] init];
    }else if(row == 3){
        vc = [[DoraemonDemoUIViewController alloc] init];
    }else if(row == 4){
        vc = [[DoraemonDemoNetViewController alloc] init];
    }else if(row == 5){
        vc = [[DoraemonDemoMockGPSViewController alloc] init];
    }else if(row == 6){
        vc = [[DoraemonDemoCrashViewController alloc] init];
    }else if(row == 7){
        vc = [[DoraemonDemoCommonViewController alloc] init];
    }else if(row == 8){
        vc = [[DoraemonDemoMemoryLeakViewController alloc] init];
    }else if(row == 9){
        vc = [[DoraemonDemoMultiControlViewController alloc] init];
    }
    if (vc) {
        [self.navigationController pushViewController:vc animated:YES];
    }
 
    //产生crash
//    NSArray *dataArray = @[@"1",@"2"];
//    NSString *num = dataArray[2];
//    NSLog(@"num == %@",num);
}

- (void)viewDidLayoutSubviews {
    [super viewDidLayoutSubviews];

    self.tableView.frame = [self fullscreen];
}

@end

File: iOS/Swift/DoKitSwiftDemo/DoKitSwiftDemo/Home/DoraemonDemoHomeViewController.swift
//
//  DoraemonDemoHomeViewController.swift
//  DoKitSwiftDemo
//
//  Created by didi on 2020/5/13.
//  Copyright © 2020 didi. All rights reserved.
//

import UIKit

class DoraemonDemoHomeViewController: DoraemonDemoBaseViewController, UITableViewDelegate, UITableViewDataSource {
    var tableView: UITableView!
    var items: [String]! = [] {
        didSet {
            self.tableView.reloadData()
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = DoraemonDemoLocalizedString("DoraemonKit")
        
        tableView = UITableView(frame: view.bounds)
        tableView.delegate = self
        tableView.dataSource = self
        view.addSubview(tableView)
        
        items  = ["沙盒测试Demo",
                 "日志测试Demo",
                 "性能测试Demo",
                 "视觉测试Demo",
                 "网络测试Demo",
                 "模拟位置Demo",
                 "crash触发Demo",
                 "通用测试Demo",
                 "内存泄漏测试",
                 "Call OC"];
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return items.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        var cell = tableView.dequeueReusableCell(withIdentifier: "HomeCellId")
        if cell == nil {
            cell = UITableViewCell(style: .default, reuseIdentifier: "HomeCellId")
        }
        cell?.textLabel?.text = DoraemonDemoLocalizedString(items[indexPath.row])
        
        return cell!;
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let row = indexPath.row
        var vc : UIViewController
        switch row {
        case 0:
            vc = DoraemonDemoSanboxViewController()
        case 1:
            vc = DoraemonDemoLoggerViewController()
        case 2:
            vc = DoraemonDemoPerformanceViewController()
        case 3:
            vc = DoraemonDemoUIViewController()
        case 4:
            vc = DoraemonDemoNetViewController()
        case 5:
            vc = DoraemonDemoGPSViewController()
        case 6:
            vc = DoraemonDemoCrashViewController()
        case 7:
            vc = DoraemonDemoCommonViewController()
        case 8:
            vc = DoraemonDemoMemoryLeakViewController()
        default:
            vc = DoraemonDemoOCViewController()
        }
        
        navigationController?.pushViewController(vc, animated: true)
    }

}
